File: Prob09/Prob09.java
import java.io.*;
import java.util.*;

public class Prob09 {
    public static void main(String[] args) {
        try {
            File f = new File("Prob09.in.txt");
            FileReader fr = new FileReader(f);
            BufferedReader br = new BufferedReader(fr);

            int n = Integer.parseInt(br.readLine());
            for(int i = 0; i < n; i++) {
                char[] message = br.readLine().toCharArray();
                char[] keyword = br.readLine().toCharArray();
                int spaces = 0;
                for(int j = 0; j < message.length; j++) {
                    if(message[j] != ' ') {
                        message[j] = (char) ((((message[j] - 'A') + (keyword[(j - spaces) % keyword.length] - 'A')) % 26) + 65);
                    } else {
                        spaces++;
                    }
                }

                System.out.println(new String(message));
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: Prob09/Prob09.py
def main():
    with open("Prob09.in.txt") as f:
        myList = [line.rstrip('\n') for line in f]
        n = int(myList[0])
        for i in range(1, n + 1):
            message = list(myList[2 * i - 1])
            keyword = list(myList[2 * i])
            spaces = 0
            for j in range(len(message)):
                if ord(message[j]) != 32:
                    message[j] = chr((((ord(message[j]) - 65) + (ord(keyword[(j - spaces) % len(keyword)]) - 65)) % 26) + 65);
                else:
                    spaces += 1
            print("".join(message))

if __name__ == "__main__": main()
